Account/FirefoxAccountConfiguration.swift

```diff
@@ -138,8 +141,83 @@ public struct ProductionFirefoxAccountConfiguration: FirefoxAccountConfiguration
     public let pushConfiguration: PushConfiguration = FirefoxPushConfiguration()
 }
 
+public struct CustomFirefoxAccountConfiguration: FirefoxAccountConfiguration {
+    
+    public init(prefs: Prefs? = nil) {
+        self.prefs = prefs
+    }
+    
+    public var prefs: Prefs?
+    
+    public let label = FirefoxAccountConfigurationLabel.custom
+    
+    public var authEndpointURL: URL {
+        get {
+            if let authServer = self.prefs?.stringForKey(PrefsKeys.KeyCustomSyncAuth), let url = URL(string: authServer + "/v1") {
+                return url
+            }
+            
+            // If somehow an invalid url was stored, fallback to the production URL
+            return ProductionFirefoxAccountConfiguration().authEndpointURL
+        }
+    }
+
+    public var oauthEndpointURL: URL {
+        get {
+            if let oauthServer = self.prefs?.stringForKey(PrefsKeys.KeyCustomSyncOauth), let url = URL(string: oauthServer + "/v1") {
+                return url
+            }
+            return ProductionFirefoxAccountConfiguration().oauthEndpointURL
+        }
+    }
+    
+    public var profileEndpointURL: URL {
+        get {
+            if let profileServer = self.prefs?.stringForKey(PrefsKeys.KeyCustomSyncProfile), let url = URL(string: profileServer + "/v1") {
+                return url
+            }
+            return ProductionFirefoxAccountConfiguration().profileEndpointURL
+        }
+    }
+    
+    public var signInURL: URL {
+        get {
+            if let signIn = self.prefs?.stringForKey(PrefsKeys.KeyCustomSyncWeb), let url = URL(string: signIn + "/signin?service=sync&context=fx_ios_v1") {
+                return url
+            }
+            return ProductionFirefoxAccountConfiguration().signInURL
+        }
+    }
+    
+    public var forceAuthURL: URL {
+        get {
+            if let forceAuth = self.prefs?.stringForKey(PrefsKeys.KeyCustomSyncWeb), let url = URL(string: forceAuth + "/force_auth?service=sync&context=fx_ios_v1") {
+                return url
+            }
+            return ProductionFirefoxAccountConfiguration().forceAuthURL
+        }
+    }
+    
+    public var settingsURL: URL {
+        get {
+            if let settings = self.prefs?.stringForKey(PrefsKeys.KeyCustomSyncWeb), let url = URL(string: settings + "/settings?service=sync&context=fx_ios_v1") {
+                return url
+            }
+            return ProductionFirefoxAccountConfiguration().settingsURL
+        }
+    }
+    
+    public var sync15Configuration: Sync15Configuration {
+        get {
+            return CustomSync15Configuration(prefs: self.prefs)
+        }
+    }
+    
+    public let pushConfiguration: PushConfiguration = FirefoxPushConfiguration()
+}
+
 public struct ChinaEditionFirefoxAccountConfiguration: FirefoxAccountConfiguration {
-    public init() {
+    public init(prefs: Prefs? = nil) {
```

Comment:
This change isn't needed if you remove the constructor from the `FirefoxAccountConfiguration`.